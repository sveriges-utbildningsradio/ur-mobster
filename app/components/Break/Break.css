.breakWrap {
  align-items: center;
  background-color: #110737;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  right: 0;
  position: absolute;
  top: 0;
  width: 70vw;
  z-index: 3;
}

.text {
  font-size: 110px;
}
