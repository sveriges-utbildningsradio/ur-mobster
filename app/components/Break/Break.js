// @flow

import React, { useContext } from 'react'
import { FormattedMessage } from 'react-intl'
import { CircularProgressbarWithChildren } from 'react-circular-progressbar'
import { SettingsStoreContext } from '../../store/store'
import { formatTime } from '../../utils/formatTime'
import styles from './Break.css'
import ResetButton from '../Button/ResetButton'

type BreakProps = {
  resetBreak: () => void
}

const Break = ({ resetBreak }: BreakProps) => {
  const { breakDuration, breakTimeLeft } = useContext(SettingsStoreContext)

  const progress = (breakTimeLeft / breakDuration) * 100

  return (
    <div className={styles.breakWrap}>
      <div>
        <CircularProgressbarWithChildren value={progress}>
          <p className={styles.text}>
            <FormattedMessage id="break" />: {formatTime(breakTimeLeft)}
          </p>
        </CircularProgressbarWithChildren>
        <ResetButton handleClick={resetBreak} id="resetBreak" />
      </div>
    </div>
  )
}

export default Break
