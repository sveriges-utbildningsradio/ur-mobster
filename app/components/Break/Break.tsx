import React, { useContext } from 'react'
import Lottie from 'react-lottie'
import { SettingsStoreContext } from '../../store/store'
import { formatTime } from '../../utils/formatTime'
import styles from './Break.css'
import ResetButton from '../Button/ResetButton'
import * as animationData from '../../assets/animations/break.json'

type BreakProps = {
  resetBreak: () => void
}

const Break = ({ resetBreak }: BreakProps) => {
  const { breakTimeLeft } = useContext(SettingsStoreContext)

  const defaultOptions = {
    loop: true,
    autoplay: true,
    animationData: animationData.default,
    rendererSettings: { preserveAspectRatio: 'xMidYMid slice' }
  }

  return (
    <div className={styles.breakWrap}>
      <div>
        <Lottie options={defaultOptions} height={500} width={500} />
        <p className={styles.text}>{formatTime(breakTimeLeft)}</p>
        <ResetButton handleClick={resetBreak} id="resetBreak" />
      </div>
    </div>
  )
}

export default Break
