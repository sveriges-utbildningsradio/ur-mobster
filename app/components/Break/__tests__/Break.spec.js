import React from 'react'
import { cleanup } from '@testing-library/react'
import 'jest-dom/extend-expect'
import renderWithReactIntl from '../../../utils/testHelpers'
import Break from '../Break'

describe('components/Break', () => {
  afterEach(cleanup)

  it('renders', () => {
    const { container } = renderWithReactIntl(<Break />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
