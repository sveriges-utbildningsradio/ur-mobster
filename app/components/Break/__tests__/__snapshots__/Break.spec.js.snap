// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Break renders 1`] = `
<div
  class="breakWrap"
>
  <div>
    <div
      data-test-id="CircularProgressbarWithChildren"
    >
      <div
        style="position: relative; width: 100%; height: 100%;"
      >
        <svg
          class="CircularProgressbar "
          data-test-id="CircularProgressbar"
          viewBox="0 0 100 100"
        >
          <path
            class="CircularProgressbar-trail"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fill-opacity="0"
            stroke-width="8"
            style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
          />
          <path
            class="CircularProgressbar-path"
            d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
            fill-opacity="0"
            stroke-width="8"
            style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
          />
        </svg>
        <div
          data-test-id="CircularProgressbarWithChildren__children"
          style="position: absolute; width: 100%; height: 100%; margin-top: -100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
        >
          <p
            class="text"
          >
            <span>
              Rast
            </span>
            : 
            05:00
          </p>
        </div>
      </div>
    </div>
    <button
      class="resetButton"
      type="button"
    >
      <span>
        Avsluta rast
      </span>
    </button>
  </div>
</div>
`;
